src/ansys/mapdl/core/mapdl_geometry.py

```diff
@@ -120,62 +157,428 @@ def _load_iges(self):
-    def _keypoints(self):
+    def _keypoints(self) -> Tuple[NDArray, NDArray]:
         """Returns keypoints cache"""
         if self._keypoints_cache is None:
             self._keypoints_cache = self._load_keypoints()
         return self._keypoints_cache
 
     @property
-    def keypoints(self):
-        """Keypoint coordinates"""
-        return np.asarray(self._keypoints.points)
+    def keypoints(self) -> pv.MultiBlock:
+        """Obtain the keypoints geometry
+
+        Obtain the selected keypoints as a :class:`pyvista.MultiBlock` object.
+
+        Returns
+        -------
+        pv.MultiBlock
+
+        Examples
+        --------
+
+        >>> mapdl.geometry.keypoints
+        MultiBlock (0x147a78220)
+          N Blocks    26
+          X Bounds    -0.016, 0.016
+          Y Bounds    -0.008, 0.018
+          Z Bounds    -0.003, 0.015
+
+        You can plot the entities:
+
+        >>> mapdl.geometry.keypoints.plot()
+
+        You can access the individual keypoints using indexing:
+
+        >>> keypoint0 = mapdl.geometry.keypoints[0]
+        >>> keypoint0
+        pyvista_ndarray([ 0.     ,  0.01778, -0.00318])
+
+        Or using the entity name:
+
+        >>> kp1 = mapdl.geometry.keypoints["kp 1"]
+        >>> kp1
+        pyvista_ndarray([ 0.     ,  0.01778, -0.00318])
+
+        You can use this to iterate over the different elements:
+
+        >>> points = mapdl.geometry.keypoints
+        >>> for each_point in points:
+                print(each_point.points[0])
+        pyvista_ndarray([ 0.     ,  0.01778, -0.00318])
+        ...
+
+        Alternatively, to iterate you can use :meth:`Mapdl.geometry.get_keypoints`.
+
+        >>> points = mapdl.geometry.get_keypoints(return_as_array=True)
+        >>> for each_point in points:
+                print(each_point)
+        [ 0.     ,  0.01778, -0.00318]
+        ...
+
+        """
+        mb = Multiblock(self.get_keypoints(return_as_list=True))
+        # Setting names
+        for ind, each_block in enumerate(mb):
+            mapdl_index = int(each_block["entity_num"][0])
+            mb.set_block_name(index=ind, name=f"kp {mapdl_index}")
+        return mb
+
+    def get_keypoints(
+        self,
+        return_as_list: bool = False,
+        return_as_array: bool = False,
+        return_ids_in_array: bool = False,
+    ) -> Union[NDArray[Any], pv.PolyData, List[pv.PolyData]]:
+        """Obtain the keypoints geometry
+
+        Obtain the selected keypoints as a :class:`pyvista.PolyData` object or
+        a list of :class:`pyvista.PolyData`
+
+        Parameters
+        ----------
+        return_as_list : bool, optional
+            Return the data as a list, by default False
+        return_as_array : bool, optional
+            Return the data as a numpy array, by default False
+        return_ids_in_array : bool, optional
+            Return the keypoints ids in the array. Only valid when
+            `return_as_array` is True. By default False
```

Comment:
```suggestion
        return_as_list : bool, optional
            Whether to return the data as a list, The default is ``False``.
        return_as_array : bool, optional
            Whether to return the data as a numpy array. The default  is ``False``.
        return_ids_in_array : bool, optional
            Whether to return the keypoint IDs in the array. The default is
            ``False``. This parameter is only valid when ``return_as_array=True``.
```
To help out the user, you should say what the behavior is when needed. For example:

 return_as_list : bool, optional
            Whether to return the data as a list, The default is ``False``, in which case
            the data is returned as a string.

Note that I'm guessing here because I don't know. Maybe our users won't know either?

Reply:
Very valid point.

Reply:
@germa89 I made some minor edits inline to your block above. I don't mind that the "in which case" is the same for two parameters in a row. At least the user knows what behavior to expect.